packages/electron-updater/src/MacUpdater.ts

```diff
@@ -79,12 +79,17 @@ export class MacUpdater extends AppUpdater {
       throw newError(`ZIP file not provided: ${safeStringifyJson(files)}`, "ERR_UPDATER_ZIP_FILE_NOT_FOUND")
     }
 
+    const provider = downloadUpdateOptions.updateInfoAndProvider.provider
+
     return this.executeDownload({
       fileExtension: "zip",
       fileInfo: zipFileInfo,
       downloadUpdateOptions,
-      task: (destinationFile, downloadOptions) => {
-        return this.httpExecutor.download(zipFileInfo.url, destinationFile, downloadOptions)
+      task: async (destinationFile, downloadOptions) => {
+        if (await this.differentialDownloadInstaller(zipFileInfo, downloadUpdateOptions, destinationFile, provider, CURRENT_MAC_APP_ZIP_FILE_NAME)) {
+          await this.httpExecutor.download(zipFileInfo.url, destinationFile, downloadOptions)
+        }
+        copyFileSync(destinationFile, this.downloadedUpdateHelper!.cacheDir + "/update.zip")
```

Comment:
I noticed this logic is different than in NSIS updater. Why do we have to copy the file if the `differentialDownloadInstaller` successfully downloads the file to the `CURRENT_MAC_APP_ZIP_FILE_NAME` already?

Reply:
NSIS copies the installer.exe to the pending directory during the installation process(https://github.com/electron-userland/electron-builder/blob/8e51ba508f39abb670bb145a79b2a3c0d4af19c9/packages/app-builder-lib/templates/nsis/include/installer.nsh#L82), and the format of the upgrade package is always .exe.  

The package format for Mac is .pkg, and the format for upgrade packages is .zip, so the only option is to copy the upgrade package to the pending directory after downloading it. 

Currently, there is a problem with this implementation: the first upgrade is always a full update, and only after the update.zip package is copied to the pending directory does it support differential updates from the second time onwards.